wgsl/extract-grammar.py

```diff
@@ -502,17 +743,196 @@ def read_spec(options):
                             last_value = None
                             result[scanner_span.name()][last_key] = []
                     else:
-                        # It's example text
+                        # A token or example
                         if scanner_parse[1] != None:
                             last_value = scanner_parse[1]
-                            result[scanner_span.name()][last_key].append(
-                                last_value)
-                    scanner_i += scanner_parse[-1]  # Advance line index
+                            if scanner_span.name() == scanner_example.name():
+                                result[scanner_span.name()][last_key].append(
+                                    last_value)
+                            if scanner_span.name() == scanner_token.name():
+                                result[scanner_span.name()][last_key] = last_value
+                    scanner_i += scanner_parse[-1]  # Advance line index 
+        for j in scanner_spans:
+            if scanner_span == j:
+                # Check if we should stop using this scanner.
+                scanner_end = j.end(scanner_lines, scanner_i)
+                if scanner_end[0]:
+                    # Yes, stop using this scanner.
+                    scanner_span = None
+                    scanner_record = False
+                    last_key = None
+                    last_value = None
+                    scanner_i += scanner_end[-1]
         scanner_i += 1
 
-    result[scanner_rule.name()]["_comment"] = [["`/\\/\\/.*/`"]]
+    bnfdialect_data = ""
+
+    # Read the contents of the bnfdialect file
+    with open(options.syntax_filename, "r") as file:
+        bnfdialect_data = file.read()
+
+    syntax_dict = {}
+    current_rule = None
+    for line in bnfdialect_data.splitlines():
+        comment = ""
+        if "//" in line:
+            line, comment = line.split("//", 1)
+        line = line.strip() # Either blankspace is insignificant
+        comment = comment.rstrip() # Right blankspace is insignificant
+        if len(line) == 0:
+            continue
+        elif current_rule is None:
+            if line.endswith(":"):
+                current_rule = line[:-1].strip()
+                syntax_dict[current_rule] = {
+                    "type": "sequence",
+                    "value": []
+                }
+        else:
+            if line.startswith(";"):
+                current_rule = None
+            else:
+                if line.startswith("|"):
+                    syntax_dict[current_rule]["type"] = "choice"
+                    line = line[1:].strip()
+                tokens = [i for i in line.split(" ") if len(i) > 0] # Strip blankspace
+                def build_production(token_j):
+                    """
+                    Takes token index and returns token steps and a (sub)production
+                    """
+                    if tokens[token_j].startswith("/"):
```

Comment:
nit: this would be easier to read by using a variable 

    token = tokens[token_j]